schnorr_fun/src/frost.rs

```diff
@@ -889,71 +889,55 @@ mod test {
                 (secret_share, frost_key)
             }).unzip();
 
-            // create a mask of signers
-            // use bytes from random u32 to determine whether a party is a signer or not
-            let mut signers_mask = vec![true];
-            while signers_mask.len() < n_parties as usize {
-                for v in signers_mask_seed.to_be_bytes() {
-                    if v > 128 {
-                        signers_mask.push(true);
-                    } else {
-                        signers_mask.push(false);
-                    }
-                    if signers_mask.len() >= n_parties as usize {
-                        break
-                    }
-                }
+            // use a boolean mask for which t participants are signers
+            let mut signer_mask = vec![true; threshold as usize];
+            signer_mask.append(&mut vec![false; (n_parties - threshold) as usize]);
+            // shuffle the mask for random signers (roughly shuffled and deterministic based on signers_mask_seed)
+            let mask_bytes = signers_mask_seed.to_be_bytes();
+            for i in 0..n_parties {
+                let rand_pos = mask_bytes[(i % mask_bytes.len() as u32) as usize] % n_parties as u8;
+                signer_mask.swap(i as usize, rand_pos as usize);
```

Comment:
I think there is a shuffle function in `rand`.

Reply:
ok cool, was just about to ask this. So OK to use rand in proptest? Using signers_mask_seed as seed for randomness

Reply:
oh good point. It's ok to use something implementing [`Rng`](https://docs.rs/rand/0.8.5/rand/seq/trait.SliceRandom.html#tymethod.shuffle) to call shuffle as long as it's deterministic. I wonder if proptest has an easy way to get the rng it uses internally? See: https://docs.rs/proptest/latest/proptest/test_runner/struct.TestRunner.html#method.rng

Reply:
I couldnt figure out how to use that `.rng` method without getting rid of the convenient `proptest!` macro. Instead i've switched to:
```
signer_mask.shuffle(&mut TestRng::deterministic_rng(RngAlgorithm::ChaCha);
```
where `TestRng::deterministic_rng(RngAlgorithm::ChaCha)` seems to provide the necessary deterministic randomness

Reply:
Actually I don't think this is doing quite what we want, it doesn't vary as the proptest runs. Trying again

Reply:
I can't figure out a nice way to get the current run context's RNG without completely scrapping the macro and doing everything manually.

Leaning towards returning to using a proptest argument for the signer mask seed. I am thinking of using a rand integer that is then used as a seed for RNG.

See this branch & line: 
https://github.com/nickfarrow/secp256kfun/commit/7d956998080a9978021886d0634b311446ce0694#diff-65e82911697a71b2673ac046e6851aa3cfa3296ad1735eb1c283f381d131fe1fR898
Random 32 bytes being fed into ChaCha seed
![image](https://user-images.githubusercontent.com/24557779/173473916-e139071b-dade-4f17-b49b-79e0e26911de.png)
(will force push here unless we can come up with a better idea!)